### DISCUSSION TITLE
Dicey Questions

### DISCUSSION THREAD

**Discussion by graylion-wa** [mDehning/wa-dice-roll#11] [Q&A] [answered]
This portion seems very straight forward...

5d10 >=7 !>=9

so the roll above will provide 1 successes for any dice results equal to or greater than 7?

and reroll any die result equal to or greater than 9?

**Dice Result Output**

Will it display the individual dice results?
this is fairly important. There are several things that can happen in systems based on the results of individual dice.
Display of the results also give a level of confidence to the user as to seeing how they arrived at the results.

8, 3, 7, 9, 2
Reroll: 8
Successes: 4

**Success & Reroll Notations**

Does it denote dice with sucesses or rerolls?

Not critical but very helpful.
could be a superscript character or assign a class for future coloring by the user.

8*, 3, 7*, 9*^, 2
Reroll: 8*
Successes: 4

**Comment by mDehning** [comment, marked as answer]
> Will it display the individual dice results?

In total, an array of dice results of the whole dice pool will be returned, in the form of `[1, 2, 3, 4, 6] = 10`

Additionally it will return both the sum of all rolls as well as the number of successes counted, if applicable.

When a dice gets rerolled/explodes, the sum of all rerolls of that exploding dice will be displayed and treated as a single dice. There is currently no display of the individual rerolls. So for example the roll of 1d10!>=8 will display `[12]` instead of `[9, 3]` or `[9 + 3]`

A rerolled Dice only counts once to the number of successes.

> Does it denote dice with sucesses or rerolls?

Dice with successes are denoted with a *. A denotation of a reroll is not realized yet, but possible.

Perhaps there is a significant distinction between "reroll" and "explode" of dice? So far I have treated those as equal

**Reply by graylion-wa** [reply]
First of all now that my work day is over,
Happy New Year!

And thank you so much for your efforts.



Now delving down with my question, it may be easier to see than say.


So using the same dice formula from above, with slightly different results.

How would the following dice results be displayed?

And would it arrive at the same 5 successes I have?


**5d10 >=7 !>=9**

**Original Roll**: 8, 3, 7, 9, 2 (3 successes) (triggers 1d reroll)
**Reroll 1**: 10 (1 success) (triggers 1d reroll)
**Reroll 2**: 8 (1 success) (trigger no more rerolls)
**Successes**: 5

**Reply by mDehning** [reply]
It would not. The result would be displayed as
[8*, 3, 7*, 27*, 2] = 3
*EDIT: I applied the rerolls on the wrong dice, I am sorry for any confusion. The First reroll on the 9 returned 10, which in turn also triggered a reroll, returning an 8*

**Reply by graylion-wa** [reply]
When counting successes (any formula with conditional targets) can we show only the indivual dice and not sum any total, only success totals.

**Reply by graylion-wa** [reply]
> First of all now that my work day is over,
> Happy New Year!
> 
> And thank you so much for your efforts.
> 
> 
> 
> Now delving down with my question, it may be easier to see than say.
> 
> 
> So using the same dice formula from above, with slightly different results.
> 
> How would the following dice results be displayed?
> 
> And would it arrive at the same 5 successes I have?
> 
> 
> **5d10 >=7 !>=9**
> 
> **Original Roll**: 8, 3, 7, 9, 2 (3 successes) (triggers 1d reroll)
> **Reroll 1**: 10 (1 success) (triggers 1d reroll)
> **Reroll 2**: 8 (1 success) (trigger no more rerolls)
> **Successes**: 5

In the above example during a reroll will additional rerolls trigger if they meet the initial reroll number of 9 or more.

Thank you

**Reply by mDehning** [reply]
Yes, you can trigger theoretically infinitive rerolls. 
I already tried to explain, the result gives BOTH the sum of the dice AS WELL as the number of successes

The String Representation ( The "[ 1, 2,3 ...] = 10" part) gives as the end result the sum of all dice, UNLESS there is at least ONE reason in the whole dice string to count successes

Please keep in mind, this is not determining in any way or shape the matter how WorldAnvil will display this. This is the underlying tool / mechanic - which does not simply return one number as a result, but a complex object that contains multiple information.

This includes
* A list of each individual roll
* A Text Representation of both the rolls and the actual result in the form of `[1, 2, 3, 4, 5, 6] = 10`
** This text will include the correct result if it is a sum - or if there are sucesses to be counted
* The actual sum of all dice
* The number of successes in the dice pool
* For good measure, the initial String that was given into the script

**Comment by mDehning** [comment]
> This portion seems very straight forward...
> 5d10 >=7 !>=9
> so the roll above will provide 1 successes for any dice results equal to or greater than 7?
> and reroll any die result equal to or greater than 9?

Yes, exactly

**Comment by mDehning** [comment]
As there are clearly at least two different interpretations of reroll vs. explode, I will open up an issue where the current notation will behave as a normal "reroll / add dice to the pool" situation and exploding dice ("roll again and add to total") will have a specific notation

**Reply by graylion-wa** [reply]
Ya exploding and reroll I think tends to be different depending on systems. I have less exposure to exploding systems and much greater to rerolls.

**Reply by graylion-wa** [reply]
Generally as soon as you counts successes you are not adding g any dice totals together, you are treating each dies value on its own from the base roll and the reroll.
